pkg/virt-controller/watch/vm.go

```diff
@@ -879,6 +880,35 @@ func (c *VMController) handleVolumeRequests(vm *virtv1.VirtualMachine, vmi *virt
 	return nil
 }
 
+func (c *VMController) handleVolumeUpdateRequest(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if !c.clusterConfig.VolumesUpdateStrategyEnabled() {
```

Comment:
If the FG is not enabled the workflow described in the proposal would "silently" break the VM, correct? Is there potential for data loss in that case?

Reply:
If and when we implement declarative volume hotplug, this new strategy could code there as well

Reply:
ACK, makes sense!

Reply:
I meant in a sense of if volumes could get lost if a user assumes the feature is enabled but in actually it isn't. (Basically what happens today?)

Reply:
Mmh not sure I understand this correctly. 
The change will be detected in any case with or without this feature gate. If this FG isn't enable then the more generic VM Restart condition will be applied.
Otherwise, the VM has the latest applied volumes, but the VMI has the current running configuration. Hence, until you reboot the VM, the volumes aren't lost, since the VMI and launcher pod still use the previous volume set.

Reply:
The replacement volume strategy is required for when we introduce new update volume strategy. Both volume migration and hotplug will cause an update of the volume set of the VM and we want to be able to distinguish between the different type of update

Reply:
Okay, understood. Maybe it is a lack of understanding on my part, but I just wanted to know what happens if you change volumes without the FG enabled. IIUC that is already caught by the VM rollout strategy and it applies a restart required condition.

Reply:
Yes, if the FG isn't enable it fallbacks to the original behavior without this feature. As I mentioned, this PR doesn't add much value, it is for breaking the changes for volume migration.